Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1434,6 +1441,53 @@ private extension ChatChannelController {
             self?.state = error == nil ? .localDataFetched : .localDataFetchFailed(ClientError(with: error))
         }
     }
+
+    private func getLastReadMessageId(firstUnreadMessageId: MessageId) -> MessageId? {
+        guard let messageIndex = messages.firstIndex(where: { $0.id == firstUnreadMessageId }) else {
+            return nil
+        }
+
+        let newLastReadMessageIndex = messages.index(after: messageIndex)
+        return messageId(at: newLastReadMessageIndex)
+    }
+
+    private func getFirstUnreadMessageId() -> MessageId? {
+        guard let lastReadMessageId = channel?.reads.first(where: { $0.user.id == client.currentUserId })?.lastReadMessageId else {
+            // We default to the oldest message in the history
+            return messages.last?.id
+        }
+
+        guard lastReadMessageId != messages.first?.id else {
+            // No unread messages
+            return nil
+        }
+        guard let lastReadIndex = messages.firstIndex(where: { $0.id == lastReadMessageId }), lastReadIndex != 0 else {
+            return nil
+        }
+
+        let lookUpStartIndex = messages.index(before: lastReadIndex)
+
+        var id: MessageId?
+        for index in (0...lookUpStartIndex).reversed() {
+            let message = message(at: index)
+            guard message?.author.id != client.currentUserId, message?.deletedAt == nil else { continue }
+            id = message?.id
+            break
+        }
```

Comment:
We will be selecting the oldest message starting from the last read that is not a message that is deleted or that is authored by the current user